lua/wire/gates/arithmetic.lua

```diff
@@ -144,12 +144,15 @@ GateActions["floor"] = {
 
 GateActions["round"] = {
 	name = "Round",
-	inputs = { "A" },
-	output = function(gate, A)
-		return math.Round(A)
+	inputs = { "A" , "B" },
+	output = function(gate, A, B)
+		if B then
+			local clampedB=math.clamp(B,-50,50)
+		end
+		return math.Round(A , clampedB or 0)
```

Comment:
`math.Round(A , B and math.clamp(B, -50, 50))`

Reply:
It will pass nil in if B is nil (which is fine), otherwise it'll pass in the clamped B

Reply:
I posted an update that just changes the original B

Reply:
I am about to do a full test in-game

Reply:
I have tested all possible variations is the script successfully works
![image](https://github.com/wiremod/wire/assets/24835664/aa8cf451-d18b-4158-ae7d-a75074aebc03)

Reply:
Un-wiring the B input makes the Rounding like the original and adding it in makes the Rounding use decimal places. Now we just need: #2760 Issue to be fixed <3

Reply:
![image](https://github.com/wiremod/wire/assets/24835664/711e848e-6d8a-4c6a-a573-b0fa8987d84a)

Reply:
On Line 152 I am seeing that I can just make two separate returns depending if I have a B or not because that B or 0 is irrelevant if I have no B

Reply:
Putting out 1 more thing just to make it look nicer I guess

Reply:
![image](https://github.com/wiremod/wire/assets/24835664/31c576cb-2b2e-4220-8a73-7d5548846e5b)